### ISSUE TITLE
The magic behind the breadcrumb

### ISSUE THREAD

**Issue by ADDISON74** [OpenMage/magento-lts#1895]
Although I've been testing Magento 1 frontends for a very long time today I noticed unusual behavior in breadcrumb. Here is how I managed to reproduce it. I used in my test environment (OpenMage 20.0.13 + Magento Sample Pack) without any extension or change in settings.

Visit all products in Sale category. On the right column in the 'Recently Viewed Products" block there are 4 visited products . If you hover your mouse over each of the links the URL's will be in this format https://mydomain.com/product-name.html.

![recently_viewd](https://user-images.githubusercontent.com/8360474/145691085-89f56b62-2e52-4108-b2e7-5a9ab7e980d1.jpg)

If you click on these links the URL in the browser window will be the one in the format shown above but the breadcrumb includes the category.

![breadcrumb-1](https://user-images.githubusercontent.com/8360474/145691095-5cfcd714-b7f8-4597-a545-7061cf360c24.jpg)

Now visit the Sale/Home & Decor category. If you click in the "Recently Viewed Product" block on the glasses the breadcrumb no longer includes a category.

![breadcrumb-2](https://user-images.githubusercontent.com/8360474/145691102-5b4dae91-37b4-4118-b07b-0ef90dc3ab4d.jpg)

Now visit Sale category and search for "sunglasses" word. Two glasses are displayed then click on Jackie O Round Sunglasses. 

![breadcrumb-3](https://user-images.githubusercontent.com/8360474/145691133-c00d63e9-6759-4ea4-9164-eba7ee3e6749.jpg)

In the product page the breadcrumb includes the category.

![breadcrumb-4](https://user-images.githubusercontent.com/8360474/145691135-13867154-754c-44d2-9da2-9cf09c687d94.jpg)

If you go to the Sale/Home & Decor category and repeat the steps on the product page the breadcrumb no longer includes the category.

![breadcrumb-5](https://user-images.githubusercontent.com/8360474/145691146-38add02d-ef4a-429e-a6a1-95eb1aeb510e.jpg)

In conclusion if you are searching for a product or it is accessed from the category where it belongs Magento adds to the breadcrumb its category in product page. If you create sub-sub-categories with products you will be surprised to find that these are added as well to the breadcrumb. Clicking in the product blocks (recently, shopping cart, wishlist, ...) or searching by accessing any product I got different breadcrumbs. This is how I managed to notice this issue.

Has anyone noticed this issue before? Obviously Magento does some tricks in the background on breadcrumb which may or may not be OK. Could we stop this "magic"?

**Comment by fballiano** [comment]
Not sure if I got everything correctly but I think I remember that the breadcrumb uses the current_category registry key to work, which is then affected by cache so, if you've caches enabled, you may get inconsistent breadcrumbs.

**Comment by ADDISON74** [comment]
I've been reading a lot about breadcrumb issues in Magento and really what's going on here is related to cache. The problem is that in the Backend in the "Cache Storage Management" section all cache types are disabled. A disable action should be done somewhere in the source code.
